### PR TITLE
adds getLocalEndpoint

### PR DESCRIPTION
**Pull request by mustafaiman**

### CODE REVIEW

**Comment by GhostInAMachine** [comment]
What purpose does this serve?

**Comment by mustafaiman** [comment]
It is useful info for users to diagnose problems, logging.

File: src/invocation/ClientConnection.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ class ClientConnection {
         this.lastRead = 0;
     }
 
+    getLocalAddress() {
+        return new Address(this.socket.localAddress, this.socket.localPort);
```
Comment (GhostInAMachine) [inline src/invocation/ClientConnection.ts]: Local address points to the address of the network interface from which the outbound connection is made, right?

Comment (mustafaiman) [inline src/invocation/ClientConnection.ts]: Yes

**Pull request merged**